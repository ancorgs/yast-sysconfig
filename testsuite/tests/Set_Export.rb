# encoding: utf-8

# Sysconfig module - testsuite
#
# Set() and Export() tests
#
# testedfiles: Sysconfig.ycp
#
# $Id$
#
module Yast
  class SetExportClient < Client
    def main
      Yast.import "Testsuite"

      Yast.import "Pkg" # override packamanager
      Yast.import "Sysconfig"

      Testsuite.Test(lambda { Sysconfig.Set(nil) }, [{}, {}, {}], nil)
      Testsuite.Test(lambda { Sysconfig.Export }, [{}, {}, {}], nil)

      Testsuite.Test(lambda { Sysconfig.Set([]) }, [{}, {}, {}], nil)
      Testsuite.Test(lambda { Sysconfig.Export }, [{}, {}, {}], nil)

      Testsuite.Test(lambda do
        Sysconfig.Set(
          [
            {
              "sysconfig_key"   => "VARIABLE",
              "sysconfig_path"  => "/etc/sysconfig/test",
              "sysconfig_value" => "no"
            }
          ]
        )
      end, [
        {},
        {},
        {}
      ], nil)
      Testsuite.Test(lambda { Sysconfig.Export }, [{}, {}, {}], nil)

      nil
    end
  end
end

Yast::SetExportClient.new.main
