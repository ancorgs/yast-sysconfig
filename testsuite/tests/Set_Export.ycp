// Sysconfig module - testsuite
//
// Set() and Export() tests
//
// testedfiles: Sysconfig.ycp
//
// $Id$
//

{

import "Testsuite";

import "Pkg";   // override packamanager

import "Mode";		// mode is imported in Product
Mode::config = true;	// package manager is initialized in Product constructor when Mode::config is false !!

import "Sysconfig";

Testsuite::Test( ``(Sysconfig::Set(nil) ), [ $[], $[], $[] ], nil );
Testsuite::Test( ``(Sysconfig::Export() ), [ $[], $[], $[] ], nil );

Testsuite::Test( ``(Sysconfig::Set([]) ), [ $[], $[], $[] ], nil );
Testsuite::Test( ``(Sysconfig::Export() ), [ $[], $[], $[] ], nil );

Testsuite::Test( ``(Sysconfig::Set([ $[ "sysconfig_key" : "VARIABLE",
    "sysconfig_path" : "/etc/sysconfig/test",
    "sysconfig_value" : "no" ] ]) ), [ $[], $[], $[] ], nil );
Testsuite::Test( ``(Sysconfig::Export() ), [ $[], $[], $[] ], nil );

}

