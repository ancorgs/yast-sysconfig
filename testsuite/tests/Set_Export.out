Return	nil
Return	[]
Return	nil
Return	[]
Return	nil
Return	[$["sysconfig_key":"VARIABLE", "sysconfig_path":"/etc/sysconfig/test", "sysconfig_value":"no"]]
